### ISSUE TITLE
Is it possible to force a specific decoder to play mp4 video?

### ISSUE THREAD

**Issue by fbs777** [mickelson/attract#236]
Is it possible to change the AM source code and force the ffmpeg to use another decoder to play mp4 (h264) video?
 I want to change from the default **h264** to **h264_mmal** on RPi2

**Comment by fbs777** [comment]
Im testing right now, and i can say this feature improves a lot the video playback.

One video that never played right is the aburner2.mp4 even at LQ (320x240), but w/ high kbps (1171 kbps + 53 kbps audio) and 60fps.

**With this aburner2.mp4 playing in software mode the CPU usage is 145% ~ 155%** ( total 4 CPU(s) @ 30% ~ 35% )

**With this aburner2.mp4 playing in mmal mode the cpu usage is 85% ~ 90%** ( total 4 CPU(s) @ 15% ~ 20% )

Now with mmal mode i can play this video without error because now every time is running lower than 100% single CPU usage

After this i tried to run a 640x480 video (60fps, 3196 kbps + 86 kbps audio). Before this patch was impossible to even start any 640x480 video, now with mmal is running @ 115% ~ 120% CPU usage, so the video run with sttutering and after ~ 10 seconds it crash, but now its just 15% ~ 20% higher than the single CPU limit.
So with this patch probably the RPI3 (im using RPI2) will run a lot of 640x480 files (not sure if all, but a lot with sure will)

 Now i will put back the mp4 files to test better and to replace flv files because with flv videos the screensaver dont show the wheel art (ffmpeg or AM bug?)

**Comment by fbs777** [comment]
Ok, after put back all 320x240 mp4 videos i got a lot of green screen as described in the commit. Sometime ago i made a mix of emumovies and progetto videosnaps by motoschito & antopisa because i was looking for the smallest videos between both packs.

Now with mmal all the green screen problem is with the progetto videosnaps videos, and all emumovies videos is working flawless.

The diff between emumovies and progetto videosnaps:

**emumovies codec: H.264 (High Profile) 

progetto videosnaps codec: H.264 (High 4:4:4 Profile)**

H.264 (High Profile) description: http://www.digitalpreservation.gov/formats/fdd/fdd000215.shtml
H.264 (High 4:4:4 Profile) description: http://www.digitalpreservation.gov/formats/fdd/fdd000218.shtml

So, basically, High Profile supports 4:2:0 chroma at 8 bits per channel and High 4:4:4 Profile supports 4:4:4 chroma at 12 bits per channel

This is not an AM problem because even with the ffplay command line the progetto videosnaps videos crash after green screen if enable mmal in the ffplay command line.

The solution is to use only emumovies videos...

**Comment by mickelson** [comment]
Thanks for the testing and reports on this, it is very helpful.

I still haven't had a chance to look at #233...
